### PR TITLE
Add spa Ids per year

### PR DESCRIPTION
**Pull request by gitsierra**
## Purpose

https://qmacbis.atlassian.net/browse/OY2-18183

Application endpoint:  https://d2rdivmid9my71.cloudfront.net/

Add Health Home Spa Ids per Year

#### Pull Request Creator Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [ ] Someone has been assigned this PR.
- [ ] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/ui-src/src/libs/spaLib.ts [hunk 1]
```diff
@@ -4,118 +4,252 @@ export interface SPAi {
...
-    name: "West Virginia Health Home for Individuals with Bipolar Disorder at Risk for Hepatitis Type B and C",
-  },
-  {
-    id: "16-0008",
-    state: "WV",
-    name: "West Virginia Health Home for Individuals with Chronic Conditions",
-  },
-  { id: "12-008", state: "WI", name: "Individuals with HIV/AIDS" },
-];
+export const SPA: any = {
```
Comment (jphechter) [inline services/ui-src/src/libs/spaLib.ts:7]:
```suggestion
export const SPA: {[year: string]: SPAi[]} = {
```
In the interest of using the SPAi type, could we do this?

File: services/ui-src/src/views/StateHome/index.tsx [hunk 2]
```diff
@@ -177,8 +177,8 @@ export const StateHome = () => {
     return <QMR.LoadingWave />;
   }
 
-  const filteredSpas = SPA.filter((s) => s.state === state);
-  const spaIds = filteredSpas.map((s) => s.id);
+  const filteredSpas = SPA[year!].filter((s: any) => s.state === state);
+  const spaIds = filteredSpas.map((s: any) => s.id);
```
Comment (jphechter) [inline services/ui-src/src/views/StateHome/index.tsx:181]:
```suggestion
  const filteredSpas = SPA[year!].filter((s: SPAi) => s.state === state);
  const spaIds = filteredSpas.map((s: SPAi) => s.id);
```
In all of the instances where we're applying `any`, could we use the SPAi type? If we specify the type in `spaLib.ts` we may actually to just remove the type.

File: services/ui-src/src/views/StateHome/index.tsx [hunk 3]
```diff
@@ -200,7 +200,7 @@ export const StateHome = () => {
   );
   const allPossibleHealthHomeCoreSetsExist = !!(
     filteredHealthHomeCoreSets.length &&
-    spaIds.every((s) =>
+    spaIds.every((s: any) =>
```
Comment (jphechter) [inline services/ui-src/src/views/StateHome/index.tsx:203]: same as above

File: services/ui-src/src/views/CoreSet/index.tsx [hunk 4]
```diff
@@ -93,12 +93,14 @@ const QualifiersStatusAndLink = ({ coreSetId }: { coreSetId: CoreSetAbbr }) => {
...
   });
-  let { state } = useParams();
+  let { state, year } = useParams();
 
   const coreSetInfo = coreSetId?.split("_") ?? [coreSetId];
   const tempSpa =
     coreSetInfo.length > 1
-      ? SPA.filter((s) => s.id === coreSetInfo[1] && s.state === state)[0]
+      ? SPA[year!].filter(
+          (s: any) => s.id === coreSetInfo[1] && s.state === state
```
Comment (jphechter) [inline services/ui-src/src/views/CoreSet/index.tsx:102]: same as in `StateHome/index.tsx`

File: services/ui-src/src/views/CoreSet/index.tsx [hunk 5]
```diff
@@ -281,7 +283,9 @@ export const CoreSet = () => {
   const coreSet = coreSetId?.split("_") ?? [coreSetId];
   const tempSpa =
     coreSet.length > 1
-      ? SPA.filter((s) => s.id === coreSet[1] && s.state === state)[0]
+      ? SPA[year].filter(
+          (s: any) => s.id === coreSet[1] && s.state === state
```
Comment (jphechter) [inline services/ui-src/src/views/CoreSet/index.tsx:287]: same as in `StateHome/index.tsx`

File: services/ui-src/src/views/AddHHCoreSet/index.tsx [hunk 6]
```diff
@@ -48,22 +48,21 @@ export const AddHHCoreSet = () => {
...
     )
-    .map((spa) => {
+    .map((spa: any) => {
       return {
         displayValue: `${spa.state} ${spa.id} - ${spa.name}`,
         value: spa.id,
       };
     })
-    .sort((a, b) => (a.displayValue > b.displayValue && 1) || -1);
+    .sort((a: any, b: any) => (a.displayValue > b.displayValue && 1) || -1);
```
Comment (jphechter) [inline services/ui-src/src/views/AddHHCoreSet/index.tsx:65]: same as in `StateHome/index.tsx`

**Review by jphechter — CHANGES_REQUESTED** [review]
Looks good overall. Not even gonna pretend that I actually read through the SPA docs on the ticket. Would like to use the `SPAi` type instead of `any`.

File: services/ui-src/src/libs/spaLib.ts [hunk 7]
```diff
@@ -4,118 +4,252 @@ export interface SPAi {
...
-  { id: "12-008", state: "WI", name: "Individuals with HIV/AIDS" },
-];
+export const SPA: { [year: string]: SPAi[] } = {
+  "2021": [
+    {
+      id: "19-0037",
+      state: "CA",
+      name: "California Health Home Program (HHP)",
+    },
+    { id: "20-0002", state: "CA", name: "California Behavioral Health Home" },
```
Comment (benjaminpaige) [inline services/ui-src/src/libs/spaLib.ts:14]: is there a reason we didn't do: 

"2021": {
CA: [
{id: "19-0037", name: "California Health Home Program (HHP)"},
{id: "20-0002", name: "California Behavioral Health Home" }
], ...

**Review by benjaminpaige — DISMISSED** [review]

**Review by jphechter — DISMISSED** [review]
LGTM 👍

**Review by benjaminpaige — APPROVED** [review]

**Pull request merged**